cfgagent/vlanconf.cpp

```diff
@@ -0,0 +1,475 @@
+    cmd += " up type bridge";
+    swss::exec(cmd, res);
+    cmd = "echo 1 > /sys/class/net/";
+    cmd += DOT1Q_BRIDGE_NAME;
+    cmd += "/bridge/vlan_filtering";
+    swss::exec(cmd, res);
+    cmd = "bridge vlan del vid " + std::to_string(DEFAULT_VLAN_ID)
+            + " dev " + DOT1Q_BRIDGE_NAME + " self";
+    swss::exec(cmd, res);
+}
+
+void VlanConf::syncCfgDB()
+{
+    OrchBase::syncDB(CFG_VLAN_TABLE_NAME, m_cfgVlanTable);
+    OrchBase::syncDB(CFG_VLAN_MEMBER_TABLE_NAME, m_cfgVlanMemberTable);
+}
+
+bool VlanConf::addHostVlan(int vlan_id)
+{
+    string cmd, res;
+
+    cmd = "bridge vlan add vid " + to_string(vlan_id) + " dev "
+            + DOT1Q_BRIDGE_NAME + " self";
+    swss::exec(cmd, res);
+    cmd = "ip link add link ";
+    cmd += DOT1Q_BRIDGE_NAME;
+    cmd += " name ";
+    cmd += VLAN_PREFIX + to_string(vlan_id)
+            + " type vlan id " + to_string(vlan_id);
+    swss::exec(cmd, res);
+
+    cmd = "ip link set ";
+    cmd += VLAN_PREFIX + to_string(vlan_id);
+    cmd += " address " + gMacAddress.to_string();
+    swss::exec(cmd, res);
+
+    // Bring up vlan port by default
+    cmd = "ip link set ";
+    cmd += VLAN_PREFIX + to_string(vlan_id);
+    cmd += " up";
+    swss::exec(cmd, res);
+    return true;
+}
+
+bool VlanConf::removeHostVlan(int vlan_id)
+{
+    string cmd, res;
+
+    cmd = "ip link del ";
+    cmd += VLAN_PREFIX + to_string(vlan_id);
+    swss::exec(cmd, res);
+
+    cmd = "bridge vlan del vid " + to_string(vlan_id) + " dev "
+            + DOT1Q_BRIDGE_NAME + " self";
+    swss::exec(cmd, res);
+
+    return true;
+}
+
+bool VlanConf::setHostVlanAdminState(int vlan_id, string &admin_status)
+{
+    string cmd, res;
+
+    cmd = "ip link set ";
+    cmd += VLAN_PREFIX + to_string(vlan_id) + " " + admin_status;
+    swss::exec(cmd, res);
+    return true;
+}
+
+bool VlanConf::setHostVlanMtu(int vlan_id, uint32_t mtu)
+{
+    string cmd, res;
+
+    cmd = "ip link set ";
+    cmd += VLAN_PREFIX + to_string(vlan_id) + " mtu " + to_string(mtu);
+    swss::exec(cmd, res);
+    return true;
+}
+
+bool VlanConf::addHostVlanMember(int vlan_id, string &port_alias, string& tagging_mode)
+{
+    string cmd, res;
+
+    // Should be ok to run set master command more than one time.
+    cmd = "ip link set " + port_alias + " master " + DOT1Q_BRIDGE_NAME;
+    swss::exec(cmd, res);
+    if (tagging_mode == "untagged" || tagging_mode == "priority_tagged")
+    {
+        // We are setting pvid as untagged vlan id.
+        cmd = "bridge vlan add vid " + to_string(vlan_id) + " dev "
+            + port_alias + " pvid untagged";
+    }
+    else
+    {
+        cmd = "bridge vlan add vid " + to_string(vlan_id) + " dev "
+            + port_alias;
+    }
+    swss::exec(cmd, res);
+    // Apply switch level flood control to this port
+    gSwtichConfVlan->updateHostFloodControl(port_alias);
+
+    // Bring up vlan member port and set MTU to 9100 by default
+    cmd = "ip link set " + port_alias + " up mtu 9100";
```

Comment:
9100 hard coded? do we have that value in config db?

Reply:
It is hard coded in interfaces.j2,  not in config DB.  Probably port schema in configDB should be updated to have MTU field.

127.0.0.1:6379[4]> hgetall "PORT|Ethernet20"
1) "alias"
2) "tenGigE20"
127.0.0.1:6379[4]> hgetall "VLAN_MEMBER|Vlan1000|Ethernet20"
1) "tagging_mode"
2) "untagged"